src/brsTypes/components/RoArray.ts

```diff
@@ -1,11 +1,42 @@
-import { BrsValue, ValueKind, BrsString, BrsBoolean, BrsInvalid } from "../BrsType";
-import { BrsType, isBrsString, isBrsNumber } from "..";
+import { BrsValue, ValueKind, BrsString, BrsBoolean, BrsInvalid, Comparable } from "../BrsType";
+import { BrsType, isBrsString, isBrsNumber, isBrsBoolean } from "..";
 import { BrsComponent, BrsIterable } from "./BrsComponent";
 import { Callable, StdlibArgument } from "../Callable";
 import { Interpreter } from "../../interpreter";
 import { Int32 } from "../Int32";
 import { RoAssociativeArray } from "./RoAssociativeArray";
 
+function sortCompare(a: BrsType, b: BrsType, caseInsensitive: boolean = false): number {
+    let compare = 0;
+    if (a !== undefined && b !== undefined) {
+        if (a instanceof RoArray && b instanceof RoAssociativeArray) {
+            compare = 1;
+        } else if (a instanceof RoAssociativeArray && b instanceof RoArray) {
+            compare = -1;
+        } else if (!(isBrsString(a) || isBrsNumber(a)) && (isBrsString(b) || isBrsNumber(b))) {
+            compare = 1;
+        } else if ((isBrsString(a) || isBrsNumber(a)) && !(isBrsString(b) || isBrsNumber(b))) {
+            compare = -1;
+        } else if (isBrsString(a) && isBrsString(b)) {
+            let aStr = a.toString();
+            let bStr = b.toString();
+            if (caseInsensitive) {
+                aStr = aStr.toLowerCase();
+                bStr = bStr.toLowerCase();
+            }
+            // roku does not use locale for sorting strings
+            compare = aStr > bStr ? 1 : -1;
+        } else if (isBrsNumber(a) && isBrsNumber(b)) {
+            compare = (a as Comparable).greaterThan(b).toBoolean() ? 1 : -1;
+        } else if (isBrsBoolean(a) && isBrsBoolean(b)) {
+            compare = 0;
+        } else {
+            compare = a > b ? 1 : -1;
```

Comment:
If we get to this point we should raise a TypeMismatch exception to match the Roku implementation. 

```
x = 3
y = "7"
if x < y (raises type mismatch error)
```

Reply:
This is no longer an issue. I discovered all other types get sorted in original order.